assets/js/app.js

```diff
@@ -0,0 +1,277 @@
+                    </div>
+                </div>
+            `)
+            .reduce((cum, cur) => cum + cur, ``)
+        document.getElementById("coinsTabPresentation").innerHTML = HTML
+
+        function allTheFunctionsForCoinPage() {
+            //make popover for all the buttons
+            const popoverTriggerList = document.querySelectorAll('[data-bs-toggle="popover"]');
+            [...popoverTriggerList].map(popoverTrigger => new bootstrap.Popover(popoverTrigger))
+
+            //load 5 coins from memory
+            const loadMemory = JSON.parse(localStorage.getItem(`5coinsInStorage`))
+            loadMemory.forEach(coin => { if (document.getElementById(coin)) document.getElementById(coin).checked = true })
+
+            //the popover button
+            const buttons = document.querySelectorAll("div > button")
+            buttons.forEach(function (button) {
+                button.addEventListener('click', async function () {
+
+                    //save the data to cache key + storage
+                    const coinsCache = JSON.parse(localStorage.getItem(`coinsCache`))
+                    const loadingDiv = document.getElementById("loadingDiv")
+                    const index = coinsCache.findIndex(coin => coin.id === this.id)
+                    let coinData = ``
+                    if (index !== -1) coinData = coinsCache[index]
+                    else {
+                        loadingDiv.classList.add(`visible`)
+                        coinData = await getSingleCoin(this.id)
+                        loadingDiv.classList.remove(`visible`)
+                    }
+
+                    if (!coinsCache.some(coin => coin.id === coinData.id)) coinsCache.push(coinData)
+                    localStorage.setItem(`coinsCache`, JSON.stringify(coinsCache))
+
+                    //after 2 min, deletes from the cache (so in the next call, get the most updated data)
+                    setTimeout(() => {
+                        const coinsCache = JSON.parse(localStorage.getItem(`coinsCache`))
+                        const index = coinsCache.findIndex(coin => coin.id = this.id)
+                        coinsCache.splice(index, 1)
+                        localStorage.setItem(`coinsCache`, JSON.stringify(coinsCache))
+                    }, 120000)
+
+                    //creates the inner html of the popover
+                    button.setAttribute('data-bs-content', `
+                    <img src="${coinData.image.thumb}" />
+                    <p>${coinData.name}</p>
+                    <p>usd: ${coinData.market_data.current_price.usd}$</p>
+                    <p>eur: ${coinData.market_data.current_price.eur}€</p>
+                    <p>ils: ${coinData.market_data.current_price.ils}₪</p>
+                `)
+                    // show the popover if there isn`t and hide if there is
+                    if (!button._popover) {
```

Comment:
this should consider also other buttons popup, so you don't up like so...
<img width="1137" alt="Screen Shot 2025-01-29 at 12 55 07" src="https://github.com/user-attachments/assets/33a1f3da-23e1-431d-84b4-377e93b68c0a" />